src/dmd/dstruct.d

```diff
@@ -456,7 +456,7 @@ extern (C++) class StructDeclaration : AggregateDeclaration
                     // CTFE sometimes creates null as hidden pointer; we'll allow this.
                     continue;
                 }
-                .error(loc, "more initializers than fields (%d) of `%s`", nfields, toChars());
+                .error(loc, "more initializers than fields (%zd) of `%s`", nfields, toChars());
```

Comment:
nfields is also `size_t` so `zu` ? (https://github.com/dlang/dmd/blob/c6ab9c400f54704a1287facc2de03d7a10c10e33/src/dmd/aggregate.d#L251)